### PR TITLE
[SDKS-7681] Add getNameByFlagSets in storages

### PR DESCRIPTION
**Pull request by nmayorsplit**
# GO SPLIT COMMONS

## What did you accomplish?
- New feature: getNamesByFlagSets in storage

## How do we test the changes introduced in this PR?
Add tests in splits_test

### CODE REVIEW

File: engine/evaluator/evaluator_test.go [hunk 1]
```diff
@@ -221,6 +221,9 @@ func (s *mockStorage) SegmentNames() *set.ThreadUnsafeSet        { return nil }
 func (s *mockStorage) SplitNames() []string                      { return make([]string, 0) }
 func (s *mockStorage) TrafficTypeExists(trafficType string) bool { return true }
 func (s *mockStorage) ChangeNumber() (int64, error)              { return 0, nil }
+func (*mockStorage) GetNamesByFlagSets(sets []string) map[string][]string {
```
Comment (mmelograno) [inline engine/evaluator/evaluator_test.go:224]: pending pattern

File: storage/inmemory/mutexmap/splits.go [hunk 2]
```diff
@@ -233,5 +233,17 @@ func (m *MMSplitStorage) TrafficTypeExists(trafficType string) bool {
 	return exists && value > 0
 }
 
+//Get flag names by sets
+func (m *MMSplitStorage) GetNamesByFlagSets(sets []string) map[string][]string {
+	toReturn := make(map[string][]string)
+	for _, flagSet := range sets {
+		flags := m.flagSets.FlagsFromSet(flagSet)
+		if len(flags) != 0 {
+			toReturn[flagSet] = flags
```
Comment (mmelograno) [inline storage/inmemory/mutexmap/splits.go:242]: sets = [] is valid return, if not it won't be possible logging the warning when set doesn't have splits linked. I would encourage to test that flow as well

**Review by mmelograno — CHANGES_REQUESTED** [review]

File: storage/inmemory/mutexmap/splits.go (hunk 2 above)
Comment (nmayorsplit) [reply storage/inmemory/mutexmap/splits.go]: Updated and added test cases to cover this.

File: engine/evaluator/evaluator_test.go (hunk 1 above)
Comment (nmayorsplit) [reply engine/evaluator/evaluator_test.go]: Updated.

**Review by mmelograno — APPROVED** [review]

**Pull request merged**